plugins/pixart-tp/fu-pxi-tp-section.c

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2025 Harris Tai <harris_tai@pixart.com>
+ * Copyright 2025 Micky Hsieh <micky_hsieh@pixart.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-pxi-tp-common.h"
+#include "fu-pxi-tp-fw-struct.h"
+#include "fu-pxi-tp-section.h"
+#include "fu-pxi-tp-struct.h"
+
+struct _FuPxiTpSection {
+	FuFirmware parent_instance;
+
+	FuPxiTpUpdateType update_type;
+	guint8 update_info;
+	guint32 flags; /* FuPxiTpFirmwareFlag bitmask */
```

Comment:
Maybe `FuPxiTpFirmwareFlags flags`? You can call it `XxxxxFlags` in the .rs file and the constants are still singular.